docs/src/tutorial.rst

```diff
@@ -0,0 +1,496 @@
+              var body = { widgets: data.Contents.map(function(e) { return e.Key }) };
+              var response = {
+                statusCode: 200,
+                headers: {},
+                body: JSON.stringify(body)
+              };
+              callback(null, response);
+            })
+            .catch(rejectedPromise);
+        }
+      }
+
+      function rejectedPromise(error) {
+        var body = error.stack || JSON.stringify(error, null, 2);
+        var response = makeError(body);
+        callback(null, response);
+      }
+    }
+
+Save it and make sure it builds and creates an empty stack.
+Note that since we haven't wired the function to our app,
+the Lambda file does not appear in the output.
+
+.. code-block:: sh
+
+    npm run build
+    cdk synth
+
+.. _create_widgets_service:
+
+Step 3: Create Widgets Service
+==============================
+
+Add the |APIGATEWAY|, |LAM|, and |S3| packages to our app.
+
+.. code-block:: sh
+
+    npm install @aws-cdk/aws-apigateway
+    npm install @aws-cdk/aws-lambda
+    npm install @aws-cdk/aws-s3
+
+Create the directory *lib* at the same level as the *bin* and *resources*
+directories.
+
+.. code-block:: sh
+
+    mkdir lib
+
+Create the following Typescript file, *widget_service.ts*,
+in the *lib* directory.
+
+.. code-block:: ts
+
+    import cdk = require('@aws-cdk/cdk');
+    import apigateway = require('@aws-cdk/aws-apigateway');
+    import lambda = require('@aws-cdk/aws-lambda');
+    import s3 = require('@aws-cdk/aws-s3');
+
+    export class WidgetService extends cdk.Construct {
+      constructor(parent: cdk.Construct, name: string) {
+        super(parent, name);
+
+        // Use S3 bucket to store our widgets
+        const bucket = new s3.Bucket(this, 'WidgetStore');
+
+        // Create a handler that calls the function main
+        // in the source file widgets(.js) in the resources directory
+        // to handle requests through API Gateway
+        const handler = new lambda.Function(this, 'WidgetHandler', {
+          runtime: lambda.Runtime.NodeJS610,
+          code: lambda.Code.directory('resources'),
+          handler: 'widgets.main',
+          environment: {
+            BUCKET: bucket.bucketName // So runtime has the bucket name
+          }
+        });
+
+        bucket.grantReadWrite(handler.role);
+
+        // Create an API Gateway REST API
+        const api = new apigateway.RestApi(this, 'widgets-api', {
+          restApiName: 'Widget Service',
+          description: 'This service serves widgets.'
+        });
+
+        // Pass the request to the handler
+        const getWidgetsIntegration = new apigateway.LambdaIntegration(handler);
+
+        // Use the getWidgetsIntegration when there is a GET request
+        api.root.addMethod('GET', getWidgetsIntegration);   // GET /
+      }
+    }
+
+Save it and make sure it builds and creates a (still empty) stack.
+
+.. code-block:: sh
+
+    npm run build
+    cdk synth
```

Comment:
What's the point of this?

Reply:
I want to reinforce that when you make changes you should rebuild and see if they are what you want.

Reply:
Fair enough